From 743b81eda124dc31dc88f355b06b857dc28d7766 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 19 Jul 2022 14:49:48 +0800
Subject: [PATCH 24/62] Revert "drm/nouveau/clk: Fix an incorrect NULL check on
 list iterator"

This reverts commit 97a9ec86ccb4e336ecde46db42b59b2ff7e0d719.
---
 drivers/gpu/drm/nouveau/nvkm/subdev/clk/base.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/nouveau/nvkm/subdev/clk/base.c b/drivers/gpu/drm/nouveau/nvkm/subdev/clk/base.c
index 5214fd0658b8..dc184e857f85 100644
--- a/drivers/gpu/drm/nouveau/nvkm/subdev/clk/base.c
+++ b/drivers/gpu/drm/nouveau/nvkm/subdev/clk/base.c
@@ -135,10 +135,10 @@ nvkm_cstate_find_best(struct nvkm_clk *clk, struct nvkm_pstate *pstate,
 
 	list_for_each_entry_from_reverse(cstate, &pstate->list, head) {
 		if (nvkm_cstate_valid(clk, cstate, max_volt, clk->temp))
-			return cstate;
+			break;
 	}
 
-	return NULL;
+	return cstate;
 }
 
 static struct nvkm_cstate *
@@ -169,8 +169,6 @@ nvkm_cstate_prog(struct nvkm_clk *clk, struct nvkm_pstate *pstate, int cstatei)
 	if (!list_empty(&pstate->list)) {
 		cstate = nvkm_cstate_get(clk, pstate, cstatei);
 		cstate = nvkm_cstate_find_best(clk, pstate, cstate);
-		if (!cstate)
-			return -EINVAL;
 	} else {
 		cstate = &pstate->base;
 	}
-- 
2.27.0

